### ISSUE TITLE
renderHeader() - Header support

### ISSUE THREAD

**Issue by xxramonxx** [Flipkart/recyclerlistview#208]
I really your project, but what i'm missing is header support. I tried to implement it on my own, and it was only a few lines of code and the header works. As I've seen in your code, header and footer are not part of an layout provider, so it should be easy to implement.
Is it possible to make render header be part of your listview ? 

Thanks a lot !

P.S.: I've experienced performance issues when i put activity indicator as footer, so i switched to simple text.

**Comment by naqvitalha** [comment]
The problem is with adjusting `distanceFromWindow` prop automatically. If you want you can pass an external scrollView with a header already attached.
Regarding the other thing, you should ideally only render `AcitivityIndicator` after getting `onEndReached` and then remove it once pagination is complete.

**Comment by phil-flyclops** [comment]
Posted a way to do something similar to this in https://github.com/Flipkart/recyclerlistview/issues/233

Could be restructured to be a HOC that takes a custom renderHeader behavior

**Comment by arunreddy-flipkart** [comment]
Closing in favour of #233

**Comment by dzpt** [comment]
#233 still doesn't work property with header @arunreddy10
